zomes/chat/src/validation.rs

```diff
@@ -78,17 +78,25 @@ pub(crate) fn joining_code(_author: AgentPubKey, membrane_proof: Option<Membrane
             let e = mem_proof.entry();
             if let ElementEntry::Present(_entry) = e {
                 let signature = mem_proof.signature().clone();
-                if verify_signature(holo_agent.clone(), signature, mem_proof.header())? {
-                    // TODO: check that the joining code has the correct author key in it
-                    // once this is added to the registration flow, e.g.:
-                    // if mem_proof.payload().agent != author {
-                    //    return Ok(ValidateCallbackResult::Invalid("Joining code invalid: incorrect agent key".to_string()))
-                    // }
-                    trace!("Joining code validated");
-                    return Ok(ValidateCallbackResult::Valid)
-                } else {
-                    trace!("Joining code validation failed: incorrect signature");
-                    return Ok(ValidateCallbackResult::Invalid("Joining code invalid: incorrect signature".to_string()))
+                match verify_signature(holo_agent.clone(), signature, mem_proof.header()) {
+                    Ok(verified) => {
+                        if verified {
+                            // TODO: check that the joining code has the correct author key in it
+                            // once this is added to the registration flow, e.g.:
+                            // if mem_proof.payload().agent != author {
+                            //    return Ok(ValidateCallbackResult::Invalid("Joining code invalid: incorrect agent key".to_string()))
+                            // }
+                            trace!("Joining code validated");
+                            return Ok(ValidateCallbackResult::Valid)
+                        } else {
+                            trace!("Joining code validation failed: incorrect signature");
+                            return Ok(ValidateCallbackResult::Invalid("Joining code invalid: incorrect signature".to_string()))
+                        }
+                    }
+                    Err(e) => {
+                        debug!("Error on get when verifying signature of agent entry: {:?}; treating as unresolved dependency",e);
+                        return Ok(ValidateCallbackResult::UnresolvedDependencies(vec![(author).into()]))
```

Comment:
Hmm is it really the `author` that's the unresolved dependency?  I think it probably doesn't matter unless holochain is doings something with the value in vec.

Reply:
That what I was thinking, I am happy to change it to anything else if you think otherwise

Reply:
It's fine for now. Lets just merge this.